File: GFG/BST/vsum.py
import ds


def pvsum(root,i,arr):
    if (root is None):
        return 0
    arr[i]+=root.key
    pvsum(root.left,i-1,arr)
    pvsum(root.right,i+1,arr)
    


root=ds.BST()
arr=[60,40,80,20,50,70,100,10,30,90,110]
for i in arr:
    root.insert(i)
root.levelorder()
print()
k=2*(root.height())-1
j=(k//2+1)
br=[0]*(k+1)
pvsum(root.root,j,br)
print(br)

File: GFG/BST/vsum.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

void vsum(BNode<int>* root,int i, vector<int> &arr){
    if(root==NULL){
        return;
    }
    arr[i]+=root->key;
    vsum(root->left,i-1,arr);
    vsum(root->right,i+1,arr);
}

int main(){
    vector<int> ins {60,40,80,20,50,70,100,10,30,90,110};
    BST<int> bt;
    for (auto i:ins){
        bt.insert(i);
    }
    int k=2*(bt.height())-1;
    vector<int> brr (k+1,0);
    int i=k/2;
    vsum(bt._root, i,brr);
    for (auto k:brr){
        cout<<k<<" ";
    }

    return 0;
}